### ISSUE TITLE
refactor "infill crtit multiplier code" for plots

### ISSUE THREAD

**Issue by berndbischl** [mlr-org/mlrMBO#169]
we have stuff like this 

code is (at least) in 
renderExampleRunPlotSingleCrit1d.R
lines 38 ff
there is a "hack" for the EI. that calculates a opt.direction -1, +1 multiplier

solution: code a minimal constructor for infill crits. 
that gather all info we want. that is better the the attributes and extensible.

makeInfillCrit(fun, minimize = FALSE, name = "expected improvment",  id = "ei")
id is used in plots for now. 
it also must match the one used in infill control. 

and we register this objects in the namespace in a list.

this shows the problem for lcb at the moment

```
#library(mlrMBO)
library(ggplot2)
library(mlrMBO)
library(smoof)
configureMlr(show.learner.output = FALSE)
pause = interactive()
set.seed(1)

obj.fun = makeSingleObjectiveFunction(
  name = "Sine",
  fn = function(x) sin(x),
  par.set = makeNumericParamSet(lower = 3, upper = 13, len = 1),
  minimize = FALSE
)

ctrl = makeMBOControl(init.design.points = 6, propose.points = 1)
ctrl = setMBOControlTermination(ctrl, iters = 10L)
ctrl = setMBOControlInfill(ctrl, crit = "lcb", opt = "focussearch", opt.focussearch.points = 500L)

lrn = makeLearner("regr.km", predict.type = "se", covtype = "matern3_2")

run = exampleRun(obj.fun, global.opt = -1, learner = lrn,
  control = ctrl, points.per.dim = 100, show.info = TRUE)

plotExampleRun(run, pause = pause, densregion = TRUE)
```

**Comment by jakobbossek** [comment]
Good point!
IMHO we should even think about offering an interface for custom infill criteria.
I already proposed this for infill optimizers (see issue #200).

**Comment by jakobbossek** [comment]
We should really do this in general and not only for internal usage. Have a look at the setMBOControlInfill function. This is just a single bloated sanity check.
I am thinking of the following solution:
```splus
library(mlrMBO)
library(smoof)
...
ctrl = makeMBOControl(init.design.points = 6, propose.points = 1)
ctrl = setMBOControlInfill(ctrl, crit.fun = setupMBOInfillEQI(ctrl, beta = 0.8))
...
``` 

and later when #200 is done even
```splus
...
ctrl = setMBOControlInfill(ctrl,
  crit.fun = setupMBOInfillEQI(ctrl, beta = 0.8)
  opt.fun = noir::setupOptimizer("focusSearch", points = 1000L)
)

**Comment by jakobbossek** [comment]
Started to work on this in branch *inill_criteria_refactoring*

**Comment by jakobbossek** [comment]
Fixed in PR #286
